packages/serverpod_shared/lib/src/config.dart

```diff
@@ -74,14 +74,30 @@ class ServerpodConfig {
 
     var doc = loadYaml(data);
 
+    /// Get api server setup. This field cannot be null, so if the
+    /// configuration is missing, create a default configuration.
     assert(doc['apiServer'] is Map, 'apiServer is missing in confing');
-    var apiServer = ServerConfig._fromJson(doc['apiServer']);
-
+    Map apiSetup = doc['apiServer'] ?? {};
+    var apiServer = apiSetup.isNotEmpty
+        ? ServerConfig._fromJson(apiSetup)
+        : ServerConfig(
+            port: 8080,
+            publicHost: 'localhost',
+            publicPort: 8080,
+            publicScheme: 'http',
+          );
+
+    /// Get insights server setup
     assert(doc['insightsServer'] is Map, 'insightsServer is missing in config');
-    var insightsServer = ServerConfig._fromJson(doc['insightsServer']);
+    Map insightsSetup = doc['insightsServer'] ?? {};
+    var insightsServer =
+        insightsSetup.isNotEmpty ? ServerConfig._fromJson(insightsSetup) : null;
```

Comment:
Suggestion: Change this to a null check instead e.g.

```dart
Map insightsSetup = doc['insightsServer'];
    var insightsServer =
        insightsSetup != null ? ServerConfig._fromJson(insightsSetup) : null;
```

This way we allow the `insightsServer` keyword to be optional BUT if you specify it you have to also give a valid config! 

This comment applies to all the other configurations in this file too.